### PR TITLE
Allow assets to be precompiled with Sprockets

### PR DESCRIPTION
**Pull request by OsamaSayegh**
Context https://github.com/MiniProfiler/rack-mini-profiler/issues/60.

This makes it possible to consume Mini Profiler's assets like any other assets in a Rails app. There is a new config option `assets_url` which is `nil` by default but can be changed to a lambda that takes 3 arguments: asset_name (i.e. `rack-mini-profiler.{js|css}`), version (which is the digest/version of MP assets), and `env` object, and the lambda should return a string that will be used as the `src` for the asset. To preserve backward compatibility, the lambda is optional and Mini Profiler will fallback to the current behavior (serve assets from `/mini-profiler-resources/*`) if no lambda is given or it returns a falsey value.

For most Rails app, the following example will work fine:

```
Rack::MiniProfiler.config.assets_url = -> (name, version, env) do
  ActionController::Base.helpers.asset_url(name)
end
```

I tested this change in a brand new Rails app and Discourse and in both apps `rake assets:precompile` generated MP assets in the `public/assets` directory, and the assets loaded just fine from the `/assets` route when I loaded a page in my browser.

Thoughts on this @SamSaffron? Is this OK? Anything you would like me to change or test?

### CODE REVIEW

**Comment by SamSaffron** [comment]
I think I am fine as long as this is 100% opt in, the PR appears to adding stuff to sprockets by default, we should only add it if people opted to sprockets management.

File: lib/mini_profiler_rails/railtie.rb [hunk 1]
```diff
@@ -6,6 +6,13 @@
 module Rack::MiniProfilerRails
   extend Rack::MiniProfilerRailsMethods
 
+  class Engine < ::Rails::Engine
```
Comment (SamSaffron) [inline lib/mini_profiler_rails/railtie.rb:9]: this whole thing should be opt in I think.

**Comment by OsamaSayegh** [comment]
Ok, I've changed this so that the engine is only created if the `assets_url` config is set. If this looks good I'll add documentation to the README file and merge it in.

**Comment by OsamaSayegh** [comment]
I'm going to merge this tomorrow unless there are objections @SamSaffron 😸

**Comment by SamSaffron** [comment]
looks good to me!

**Pull request merged**